src/frontend/screens/Settings/MapManagement/BackgroundMaps.tsx

```diff
@@ -0,0 +1,364 @@
+              {t(m.removeMapFile)}
+            </Text>
+          </Button>
+        </>
+      );
+
+      break;
+    }
+    case 'success': {
+      if (customMapInfoQuery.isFetching) {
+        renderedMapInfo = customMapInfoQuery.data ? (
+          <CustomMapDetails
+            loading
+            name={customMapInfoQuery.data.name}
+            dateAdded={customMapInfoQuery.data.created}
+            size={customMapInfoQuery.data.size}
+            onRemove={() => {
+              removeMapBottomSheet.openSheet();
+            }}
+          />
+        ) : (
+          <Loading size={10} />
+        );
+      } else {
+        renderedMapInfo = customMapInfoQuery.data ? (
+          <CustomMapDetails
+            name={customMapInfoQuery.data.name}
+            dateAdded={customMapInfoQuery.data.created}
+            size={customMapInfoQuery.data.size}
+            onRemove={() => {
+              removeMapBottomSheet.openSheet();
+            }}
+          />
+        ) : (
+          <ChooseMapFile
+            onChooseFile={() => {
+              selectCustomMapMutation.mutate(undefined, {
+                onSuccess: asset => {
+                  if (!asset) return;
+
+                  importCustomMapMutation.mutate(
+                    {
+                      uri: asset.uri,
+                    },
+                    {
+                      onError: () => {
+                        FileSystem.deleteAsync(asset.uri, {
+                          idempotent: true,
+                        }).catch(noop);
+                      },
+                      onSuccess: () => {
+                        mapAddedBottomSheet.openSheet();
+                      },
+                    },
+                  );
+                },
+              });
+            }}
+          />
+        );
+      }
+
+      break;
+    }
+  }
+  return (
+    <>
+      <ScrollView contentContainerStyle={styles.container}>
+        <Text style={styles.aboutText}>{t(m.about)}</Text>
+        <View style={styles.descriptionContainer}>
+          <Text>{t(m.description1)}</Text>
+          <Text>{t(m.description2)}</Text>
+        </View>
+        {renderedMapInfo}
+      </ScrollView>
+
+      <BottomSheetModal
+        ref={removeMapBottomSheet.sheetRef}
+        isOpen={removeMapBottomSheet.isOpen}>
+        <BottomSheetModalContent
+          icon={<ErrorSvg />}
+          title={t(m.deleteCustomMapTitle)}
+          description={
+            t(m.deleteCustomMapDescription) + '\n\n' + t(m.cannotBeUndone)
+          }
+          buttonConfigs={[
+            {
+              dangerous: true,
+              variation: 'filled',
+              text: t(m.deleteMapButtonText),
+              icon: <MaterialIcon size={30} name="delete" color={WHITE} />,
+              onPress: () => {
+                removeCustomMapMutation.mutate();
+                removeMapBottomSheet.closeSheet();
+              },
+            },
```

Comment:
in this case i would argue against it for the following reasons:

1. One is a full-sized sheet and the other isn't. If i remember correctly, we had issues with dynamically switching between full-size and non-full-size (some issue with the component implementation)

2. It provides a more explicit separation of states that can occur. Part of this is due to a limitation of the how the bottom sheet component is designed, but i'd rather avoid code that does something like `isOpen={mapAddedOpen || mapRemovedOpen}`, because then there's a greater risk of introducing or seeing the wrong sheet content, unless you're careful about the checks you make in it.